multicast/exceptions.py

```diff
@@ -0,0 +1,271 @@
+	baton.module = __module__
+	baton.path = __file__
+	baton.__cause__ = err
+	raise baton from err
+
+
+class CommandExecutionError(RuntimeError):
+	"""
+	Exception raised when a command execution fails.
+
+	Attributes:
+		message (str) -- Description of the error.
+		exit_code (int) -- The exit code associated with the error.
+
+	Meta-Testing:
+
+		Testcase 1: Initialization with message and exit code.
+			A. - Initializes the error.
+			B. - checks inheritance.
+			C. - checks each attribute.
+
+			>>> error = CommandExecutionError("Failed to execute command", 1)
+			>>> isinstance(error, RuntimeError)
+			True
+			>>> error.message
+			'Failed to execute command'
+			>>> error.exit_code
+			1
+	"""
+
+	def __init__(self, *args, **kwargs):
+		"""
+		Initialize CommandExecutionError with a message and exit code.
+
+		Parameters:
+			message (str) -- Description of the error.
+			exit_code (int) -- The exit code associated with the error.
+			*args: Variable length argument list.
+			**kwargs: Arbitrary keyword arguments.
+
+		Raises:
+			TypeError: if the exit_code is not an int.
+
+		Meta-Testing:
+
+			Testcase 1: Initialization with different exit code:
+				A. - Initializes a CommandExecutionError with a specific exit code.
+				B. - Checks the message is still set, as super class would.
+				C. - check the specific exit code is 2.
+
+				>>> error = CommandExecutionError("Error message", 2)
+				>>> error.message
+				'Error message'
+				>>> error.exit_code
+				2
+		"""
+		if len(args) > 0 and isinstance(args[-1], int):
+			exit_code = args[-1]
+			args = args[:-1]
+		else:
+			exit_code = kwargs.pop("exit_code", 1)
+		super().__init__(*args, **kwargs)
+		self.message = args[0] if args else kwargs.get("message", "An error occurred")
+		self.exit_code = exit_code
+
+
+EXIT_CODES = {
+	0: (None, 'Success'),
+	1: (RuntimeError, 'General Error'),
+	2: (OSError, 'Misuse of Shell Builtins'),
+	64: (argparse.ArgumentError, 'Usage Error'),
+	65: (ValueError, 'Data Error'),
+	66: (FileNotFoundError, 'No Input'),
+	69: (ConnectionError, 'Unavailable Service'),
+	70: (Exception, 'Internal Software Error'),
+	77: (PermissionError, 'Permission Denied'),
+	125: (BaseException, 'Critical Failure'),
+	126: (AssertionError, 'Command Invoked Cannot Execute'),
+	127: (ModuleNotFoundError, 'Command Not Found'),
+	129: (None, 'Hangup (SIGHUP)'),
+	130: (KeyboardInterrupt, 'Interrupt (SIGINT)'),
+	134: (None, 'Abort (SIGABRT)'),
+	137: (None, 'Killed (SIGKILL)'),
+	141: (BrokenPipeError, 'Broken Pipe (SIGPIPE)'),
+	143: (SystemExit, 'Terminated (SIGTERM)'),
+	255: (None, 'Exit Status Out of Range'),
+}
+
+
+EXCEPTION_EXIT_CODES = {exc: code for code, (exc, _) in EXIT_CODES.items() if exc}
+
+
+def get_exit_code_from_exception(exc):
+	for exc_class in EXCEPTION_EXIT_CODES:
+		if isinstance(exc, exc_class):
+			return EXCEPTION_EXIT_CODES[exc_class]
+	return 70  # Default to 'Internal Software Error'
+
+
+def exit_on_exception(func):
```

Comment:
@coderabbitai, Please read the code for the function `exit_on_exception` and the GH issue #117 and generate a CEP-7 compliant docstrings

Reply:
```suggestion
def exit_on_exception(func):
	"""
	Decorator that wraps a function to handle exceptions and exit with appropriate exit codes.

	This decorator captures exceptions raised by the wrapped function and handles them by mapping them to predefined exit codes specified in `EXIT_CODES`. It ensures that both `SystemExit` exceptions (which may be raised by modules like `argparse`) and other base exceptions result in the program exiting with meaningful exit codes and error messages.

	Parameters:
		func (callable): The function to be wrapped by the decorator.

	Returns:
		callable: The wrapped function with enhanced exception handling.

	Meta-Testing:

		Testcase 1: Successful execution without exceptions.
			A. Define a function that returns a value.
			B. Decorate it with `@exit_on_exception`.
			C. Call the function and confirm it returns the expected value.

			>>> @exit_on_exception
			... def sample_func():
			...     return "Success"
			>>> sample_func()
			'Success'

		Testcase 2: Function raises `SystemExit` exception.
			A. Define a function that raises `SystemExit`.
			B. Decorate it with `@exit_on_exception`.
			C. Call the function and verify it exits with the correct exit code.

			>>> @exit_on_exception
			... def system_exit_func():
			...     raise SystemExit(64)
			>>> system_exit_func()  # Exits with code 64

		Testcase 3: Function raises a general exception.
			A. Define a function that raises a `ValueError`.
			B. Decorate it with `@exit_on_exception`.
			C. Call the function and verify it exits with the mapped exit code.

			>>> @exit_on_exception
			... def error_func():
			...     raise ValueError("Invalid value")
			>>> error_func()  # Exits with code 65

	"""
```